lib/fetch/headers.js

```diff
@@ -241,6 +241,8 @@ class HeadersList {
    * @param {string} value
    */
   lowerCaseSet (name, value) {
+    this[kHeadersSortedMap] = null
```

Comment:
I'm sorry. I deleted it by mistake.

Reply:
Are we missing a test that can cache this?

Reply:
No, I forgot to add here. https://github.com/nodejs/undici/pull/2516

Reply:
> Are we missing a test that can catch this?

Should I be adding tests?

Reply:
There seems to be no HeadersList testing.

Reply:
Please 🙏

Reply:
why do we need lowercaseGet/lowercaseDelete? Problems like these occur when we deviate from the spec...

Reply:
> why do we need lowercaseGet/lowercaseDelete? Problems like these occur when we deviate from the spec...

As shown in #2516, too many toLowerCase are called.
A simple fetch would be called at least 30 times.

Reply:
I understand that, but I don't understand why they have to be separate. As in, why can't lowerCaseAppend just be append?

Reply:
Another advantage is that this method, added in #2501, can be used.

Reply:
> As in, why can't lowerCaseAppend just be append?

I didn't do that because I was afraid I might accidentally call it that. Even though it's capitalized.

Reply:
> As in, why can't lowerCaseAppend just  be append?

I tried to change the append, I didn't do that because I was afraid I might accidentally call it that. Even though it's capitalized.

Reply:
if append is identical to lowerCaseAppend, why are you afraid of calling it that...? it's following the steps from the spec? What's capitalized?

Reply:
Okay I disagree heavily on implementation but I understand what it's doing. Maintaining 2 versions of all of these is not maintainable and should be removed. Instead of the `lowerCase_` methods, why not add a third parameter to them called "isLowercase"/something along the lines of that (and then handling casing there)? It's closer to the spec and much easier to maintain that way.

Reply:
Okey I'll do it that way.